src/core/dbus-unit.c

```diff
@@ -393,16 +393,21 @@ int bus_unit_method_start_generic(
         else
                 verb = job_type_to_string(job_type);
 
+        /* We don't want to propagate the error, so use a different error struct here */
+        _cleanup_(sd_bus_error_free) sd_bus_error e = SD_BUS_ERROR_NULL;
+
         r = bus_verify_manage_units_async_full(
                         u,
                         verb,
                         CAP_SYS_ADMIN,
                         polkit_message_for_job[job_type],
                         true,
                         message,
-                        error);
-        if (r < 0)
-                return r;
+                        &e);
+        if (r < 0) {
+                log_debug_errno(r, "Polkit authorization failed: %s", bus_error_message(&e, r));
+                return sd_bus_error_set(error, SD_BUS_ERROR_ACCESS_DENIED, "Permission denied");
```

Comment:
i am pretty sure this should be placed in the polkit code, i.e. in bus_verify_polkit_async(). In fact there's already some code in place to eat up certain polkit errors and turning them into plain EACCES, it just needs to cover more, most importanltly the "interactive authentication needed one".

The patch becomes a three-liner then I guess.

Reply:
Duh, for some reason I didn't think of this. Updated.